src/services/dom/helpers.ts

```diff
@@ -90,3 +90,36 @@ export const preorder = (
   }
   return acc;
 };
+
+/**
+ * Attempt to find an element by id in the given node
+ * Handle cases where an element is passed in instead of a document
+ */
+export const getElementById = (
+  doc: Document | null | undefined,
+  id: string,
+): Element | null | undefined => {
+  if (!doc) {
+    return doc;
```

Comment:
I assume here Document would be `null` or `undefined`, but I wonder if that makes sense to return "doc". Should we explicitly return `null` instead?